tests/test_aaa_authorization_service.rb

```diff
@@ -0,0 +1,1230 @@
+
+# TestAaaAuthorizationService - Minitest for AaaAuthorizationService util
+# rubocop:disable ClassLength
+class TestAaaAuthorizationService < CiscoTestCase
+  # Method to pre-configure a valid tacacs server and aaa group.  This
+  # group can be included in the testing such access to the device
+  # never is compromised.
+  def preconfig_tacacs_server_access(group_name, keep=true)
+    @device.cmd('configure terminal')
+    if keep
+      @device.cmd('tacacs-server key testing123')
+      @device.cmd('tacacs-server host 10.122.197.197 key testing123')
+      @device.cmd("aaa group server tacacs+ #{group_name}")
+      @device.cmd('server 10.122.197.197')
+      @device.cmd('use-vrf management')
+      @device.cmd('source-interface mgmt0')
+      @device.cmd('aaa authentication login ascii-authentication')
+    else
+      @device.cmd("no aaa group server tacacs+ #{group_name}")
+    end
+    @device.cmd('end')
+    # Flush the cache since we've modified the device outside of RLB
+    node.cache_flush
+  end
+
+  def feature_tacacs(feature=true)
+    @device.cmd('configure terminal')
+    if feature
+      @device.cmd('feature tacacs')
+    else
+      @device.cmd('no feature tacacs')
+      @device.cmd('no aaa authentication login ascii-authentication')
+    end
+    @device.cmd('end')
+    # Flush the cache since we've modified the device outside of RLB
+    node.cache_flush
+  end
+
+  def config_tacacs_servers(servers)
+    @device.cmd('configure terminal')
+    @device.cmd('feature tacacs+')
+    servers.each do |server|
+      @device.cmd("aaa group server tacacs+ #{server}")
+    end
+    @device.cmd('end')
+    # Flush the cache since we've modified the device outside of RLB
+    node.cache_flush
+  end
+
+  def get_match_line(name)
+    s = @device.cmd('show run aaa all | no-more')
+    prefix = 'aaa authorization'
+    line = /#{prefix} #{name}/.match(s)
+    line
+  end
+
+  def test_aaaauthorizationservice_create_unsupported_type
+    assert_raises(ArgumentError) do
+      AaaAuthorizationService.new(:none, 'default')
+    end
+  end
+
+  def test_aaaauthorizationservice_create_nil_type
+    assert_raises(TypeError) do
+      AaaAuthorizationService.new(nil, 'default')
+    end
+  end
+
+  def test_aaaauthorizationservice_create_invalid_type
+    assert_raises(TypeError) do
+      AaaAuthorizationService.new('test', 'default')
+    end
+  end
+
+  def test_aaaauthorizationservice_create_invalid_range_type
+    assert_raises(TypeError) do
+      AaaAuthorizationService.new(34, 'default')
+    end
+  end
+
+  def test_aaaauthorizationservice_create_invalid_service
+    assert_raises(ArgumentError) do
+      AaaAuthorizationService.new(:commands, 'test')
+    end
+  end
+
+  def test_aaaauthorizationservice_create_empty_service
+    assert_raises(ArgumentError) do
+      AaaAuthorizationService.new(:commands, '')
+    end
+  end
+
+  def test_aaaauthorizationservice_create_commands_default
+    feature_tacacs
```

Comment:
Consider defining setup and teardown methods to avoid repeated code in each test, i.e.:

```
def setup
  super
  feature_tacacs
end

def teardown
  feature_tacacs(false)
  super
end
```